### ISSUE TITLE
Are TensorRT inference times with multiple streams essentially a sum?

### ISSUE THREAD

**Issue by geiche735** [ultralytics/ultralytics#14924]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am doing TensorRT inference on 14 camera streams. While doing inference I am seeing inference times of such as the following:
0: 2.0 ms, 1: 2.0 ms, ...., 13: 2.0 ms

This is one frame of the inference for each camera. Is the total inference time a sum of the 2 ms for every camera (2*14 = 28 ms), or do they all perform inference in parallel in the same 2 ms (total is 2 ms for all 14 cams)?

**Comment by Burhan-Q** [comment]
Inference times are the total time to run inference on the entire batch divided by the number of images in a batch

https://github.com/ultralytics/ultralytics/blob/9f22f45189bd32efc14ddcb7b7b42a548b7aef6f/ultralytics/engine/predictor.py#L265-L272

Batch inference is synchronous and not asynchronous. I've started to look into how to do this, but open to any ideas or recommendations. My aim is to try to figure out how to execute async inference **without** introducing a new dependency to the repo tho, which makes it a bit trickier.

**Comment by geiche735** [comment]
Thanks for your response. I accidentally provided the wrong result, the result from standard inference, not TensorRT. For TRT I was seeing a similar output to the code you posted, with inference:  2.0 ms. So say I have 14 cameras, the total inference time would be 28 ms? I am a little confused as you said inference was synchronous, so would all inference take a total time of 28 ms (one image processed, then the next, then the next), or synchronously (all images in batch processed in parallel) in 2 ms?

**Comment by Burhan-Q** [comment]
@geiche735 in the code I cited above, the total inference time (in milliseconds) is defined by `profilers[1].dt * 1e3` this would be the case for one image, 16 images, or three images. The "per-image" inference time that is reported is the total inference time, divided by the total number of images in a batch. So if there are 14 images in your batch and the per-image inference time is 2.0 ms for each, then the total inference time is 28.0 ms. 
I had thought previously that the value reported was total inference time for the batch (in your case 2.0 ms for all images), but after reviewing the code from your question, that's not the case. Essentially the "per-image" inference time is an approximate average of inference time for each image in the batch. The inference call for TRT is https://github.com/ultralytics/ultralytics/blob/5d9046abda3d1070bc2c4bf234a1ab81f50e43e5/ultralytics/nn/autobackend.py#L516 which uses [synchronous inference](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/ExecutionContext.html#tensorrt.IExecutionContext.execute_v2), as the alternative would be [execute_async_v3](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/ExecutionContext.html#tensorrt.IExecutionContext.execute_async_v3) (but would additionally require more setup). As I mentioned earlier, I've looked into async execution some, but haven't progressed much with it as of yet.
